From 47fc6842a7604a2a1d7478ab26a3ecfdf8497d45 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Thu, 18 Apr 2013 10:58:26 +0800
Subject: [PATCH 189/602] ENGR00258713 BT: enable NAP to share network to
 others

1.delete PAN_NAP_DISABLED to enable BT newwork access point.
2.enbale BT NAP UI through config_tether_bluetooth_regexs.
3.clear some minor tweaks of BT config in imx5x and imx6sabresd.

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 imx53_smd/bluetooth/bdroid_buildcfg.h               |  1 -
 .../frameworks/base/core/res/res/values/config.xml  | 21 +++------------------
 sabresd_6dq/bluetooth/bdroid_buildcfg.h             |  1 -
 .../frameworks/base/core/res/res/values/config.xml  | 17 ++++++-----------
 4 files changed, 9 insertions(+), 31 deletions(-)

diff --git a/imx53_smd/bluetooth/bdroid_buildcfg.h b/imx53_smd/bluetooth/bdroid_buildcfg.h
index 7e9890e..eba03f7 100644
--- a/imx53_smd/bluetooth/bdroid_buildcfg.h
+++ b/imx53_smd/bluetooth/bdroid_buildcfg.h
@@ -28,6 +28,5 @@
 
 #define BTIF_HF_SERVICES (BTA_HSP_SERVICE_MASK)
 #define BTIF_HF_SERVICE_NAMES  { BTIF_HSAG_SERVICE_NAME }
-#define PAN_NAP_DISABLED TRUE
 
 #endif
diff --git a/imx53_smd/overlay/frameworks/base/core/res/res/values/config.xml b/imx53_smd/overlay/frameworks/base/core/res/res/values/config.xml
index d8c12df..36e353e 100755
--- a/imx53_smd/overlay/frameworks/base/core/res/res/values/config.xml
+++ b/imx53_smd/overlay/frameworks/base/core/res/res/values/config.xml
@@ -37,28 +37,11 @@
       <item>"wlan0"</item>
     </string-array>
 
-    <!-- Regex array of allowable upstream ifaces for tethering - for example if you want
-         tethering on a new interface called "foo2" add <item>"foo\\d"</item> to the array -->
-    <add-resource type="string" name="config_tether_upstream_regexs">
-       <item>ppp.*</item>
-       <item>wlan.*</item>
-    </add-resource>
-
-    <!-- Array of ConnectivityManager.TYPE_xxxx values allowable for tethering -->
-    <!-- Common options are [1, 4] for TYPE_WIFI and TYPE_MOBILE_DUN or
-    <!== [0,1,5,7,9] for TYPE_MOBILE, TYPE_WIFI, TYPE_MOBILE_HIPRI, TYPE_BLUETOOTH and TYPE_ETHERNET -->
-    <integer-array translatable="false" name="config_tether_upstream_types">
-        <item>0</item>
-        <item>1</item>
-        <item>5</item>
-        <item>9</item>
-    </integer-array>
-
     <!-- List of regexpressions describing the interface (if any) that represent tetherable
          bluetooth interfaces.  If the device doesn't want to support tethering over bluetooth this
          should be empty. -->
     <string-array translatable="false" name="config_tether_bluetooth_regexs">
-        <item>"bnep\\d"</item>
+        <item>"bt-pan"</item>
     </string-array>
 
     <!-- An Array of "[ConnectivityManager connectionType],
@@ -66,6 +49,7 @@
     <string-array translatable="false" name="radioAttributes">
         <item>"1,1"</item>
         <item>"0,1"</item>
+        <item>"7,1"</item>
         <item>"9,1"</item>
     </string-array>
 
@@ -85,6 +69,7 @@
         <item>"mobile_ims,11,0,3,60000,true"</item>
         <item>"mobile_cbs,12,0,3,60000,true"</item>
         <item>"wifi_p2p,13,1,0,-1,true"</item>
+        <item>"bluetooth,7,7,2,-1,true"</item>
         <item>"ethernet,9,9,2,-1,true"</item>
     </string-array>
 
diff --git a/sabresd_6dq/bluetooth/bdroid_buildcfg.h b/sabresd_6dq/bluetooth/bdroid_buildcfg.h
index 20757ee..d6f4c4b 100644
--- a/sabresd_6dq/bluetooth/bdroid_buildcfg.h
+++ b/sabresd_6dq/bluetooth/bdroid_buildcfg.h
@@ -28,6 +28,5 @@
 
 #define BTIF_HF_SERVICES (BTA_HSP_SERVICE_MASK)
 #define BTIF_HF_SERVICE_NAMES  { BTIF_HSAG_SERVICE_NAME }
-#define PAN_NAP_DISABLED TRUE
 
 #endif
diff --git a/sabresd_6dq/overlay/frameworks/base/core/res/res/values/config.xml b/sabresd_6dq/overlay/frameworks/base/core/res/res/values/config.xml
index e6ecf89..9975208 100755
--- a/sabresd_6dq/overlay/frameworks/base/core/res/res/values/config.xml
+++ b/sabresd_6dq/overlay/frameworks/base/core/res/res/values/config.xml
@@ -34,17 +34,12 @@
     <string-array translatable="false" name="config_tether_wifi_regexs">
       <item>"wlan0"</item>
     </string-array>
-
-    <!-- Array of ConnectivityManager.TYPE_xxxx values allowable for tethering -->
-    <!-- Common options are [1, 4] for TYPE_WIFI and TYPE_MOBILE_DUN or
-    <!== [0,1,5,7,9] for TYPE_MOBILE, TYPE_WIFI, TYPE_MOBILE_HIPRI, TYPE_BLUETOOTH and TYPE_ETHERNET -->
-    <integer-array translatable="false" name="config_tether_upstream_types">
-        <item>0</item>
-        <item>1</item>
-        <item>5</item>
-        <item>7</item>
-        <item>9</item>
-    </integer-array>
+    <!-- List of regexpressions describing the interface (if any) that represent tetherable
+         bluetooth interfaces.  If the device doesn't want to support tethering over bluetooth this
+         should be empty. -->
+    <string-array translatable="false" name="config_tether_bluetooth_regexs">
+      <item>"bt-pan"</item>
+    </string-array>
 
     <!-- Flag indicating whether we should enable the automatic brightness in Settings.
          config_hardware_automatic_brightness_available is not set, so we will use software implementation -->
-- 
1.8.0

